### PR TITLE
[Plugin] Add plugin in c interface

### PR DESCRIPTION
**Pull request by ibmibmibm**

### CODE REVIEW

File: include/api/wasmedge/wasmedge.h [hunk 1]
```diff
@@ -138,6 +144,65 @@ typedef struct WasmEdge_Async WasmEdge_Async;
...
+  const char *Description;
+  uint32_t APIVersion;
+  WasmEdge_PluginVersionData Version;
+  uint32_t ModuleCount;
+  uint32_t ProgramOptionCount;
+  WasmEdge_ModuleDescriptor *ModuleDescriptions;
+  WasmEdge_ProgramOption *ProgramOptions;
+} WasmEdge_PluginDescriptor;
+
+#define WasmEdge_Plugin_CurrentAPIVersion UINT32_C(2)
```
Comment (hydai) [inline include/api/wasmedge/wasmedge.h:204]: Should we use cmake to define this number? I found that this version will be used in both `wasmedge.h` and `plugin/plugin.h`.

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by q82419** [comment]
Should it have the deleter in `WasmEdge_ModuleDescriptor`?

File: include/api/wasmedge/wasmedge.h (hunk 1 above)
Comment (ibmibmibm) [reply include/api/wasmedge/wasmedge.h]: Moved.

**Review by hydai — APPROVED** [review]
LGTM

**Review by hydai — APPROVED** [review]

**Review by q82419 — APPROVED** [review]

**Pull request merged**